### PR TITLE
chore: standardize test setup across Python instrumentors

### PR DESCRIPTION
**Pull request by caroger**
### Summary
- Drop Python 3.9 support and update CI/envlists to 3.10+ only.
- Standardize test installs and recording behavior across instrumentors; make pinned vs latest combos deterministic.
- Update multiple instrumentors and re‑record cassettes to match upstream API changes.
### Key Changes

- tox.ini now targets 3.10+ and exercises pinned vs latest versions consistently.
- Added broader passenv coverage for API keys used in recording.
- pytest-recording is the preferred recorder; pytest-vcr is explicitly removed for google‑genai envs.
- Updated/normalized test requirements, conftest configs, and cassettes across OpenAI, Mistral, LiteLLM, Smolagents, Pydantic AI, Autogen‑agentchat, LlamaIndex, Haystack, Groq, etc.

Misc

- .gitignore now ignores **/.mypy_cache.
- AGENTS.md updated to reflect Python 3.10+.

---

> [!NOTE]
> **Medium Risk**
> Mostly test/configuration refactoring, but it also changes import timing and type handling inside several production instrumentors (notably LiteLLM), which could alter runtime behavior or surface new edge cases with optional dependencies.
> 
> **Overview**
> Drops Python 3.9 across the Python instrumentation packages (docs and multiple `pyproject.toml` `requires-python` / classifiers), and updates repo hygiene (ignore `**/.mypy_cache`).
> 
> Standardizes testing/recording: removes `test` optional-dependency groups from many instrumentors in favor of per-package `test-requirements.txt`, switches remaining `pytest-vcr` usage to `pytest-recording`, and centralizes VCR behavior via new/updated `conftest.py` fixtures that strip headers and set `record_mode="once"` (plus a custom matcher for Google ADK).
> 
> Improves runtime robustness/compatibility by deferring imports of optional third-party SDKs and wrapper classes until `_instrument()`/`instrument()` time in several instrumentors (e.g., `agno`, `groq`, `instructor`, `autogen`), and updates LiteLLM instrumentation to support newer LiteLLM versions (bumping min to `>=1.65.0`) via extensive lazy-import/type loosening changes and updated test deps/cassettes.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit acf1f2cd3777a1545113acb30907a6f97679cff0. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: python/instrumentation/openinference-instrumentation-google-adk/tests/conftest.py [hunk 1]
```diff
@@ -20,16 +20,26 @@ def _strip_response_headers(response: Any) -> Any:
...
         "before_record_response": _strip_response_headers,
         "decode_compressed_response": True,
         "record_mode": "once",
+        "match_on": ["method_case_insensitive", "scheme", "host", "port", "path", "query"],
+        "custom_matchers": {"method_case_insensitive": _match_method_case_insensitive},
     }
 
 
+def pytest_recording_configure(config: Any, vcr: Any) -> None:
+    vcr.register_matcher("method_case_insensitive", _match_method_case_insensitive)
```
[the remark this answers is not in the document]
Comment (caroger) [reply python/instrumentation/openinference-instrumentation-google-adk/tests/conftest.py:40]: `pytest-recording` does define and invoke `pytest_recording_configure`: it’s a hookspec in `pytest_recording/hooks.py` and is called when the VCR instance is created in `pytest_recording/_vcr.py` (right before `use_cassette`). That means the matcher is registered before `match_on` is resolved, so we won’t hit the unknown‑matcher error.
